src/Qir/Tests/FullstateSimulator/FullstateSimulatorTests.cpp

```diff
@@ -145,6 +145,10 @@ TEST_CASE("Fullstate simulator: ZZ measure", "[fullstate_simulator]")
     Result rOne = iqa->Measure(2, paulis, 2, q);
     REQUIRE(Result_One == sim->GetResultValue(rOne));
 
+    iqa->X(q[1]);
+    iqa->ControlledX(1, &q[0], q[1]);
+    iqa->H(q[0]);
```

Comment:
Looks like some more comments would not harm in this test. I'm not sure my understanding is correct about the added lines here. Here is how I interpret the logic:

After line 138 the state of the two allocated qubits is `|00>` (the qubit `q[0]` is on the left).
After 139 the `q[0]` is |+>, equally probable |0> or |1> in PauliZ basis. The state of two qubits is equally probable |00> or |10> (in PauliZ).
After 140 the state is equally probable |00> or |11> (Bell state) (in PauliZ).
After 141 the state collapses to either |00> or |11>.

After 144 the state is either |01> or |10>. 
After 145 the state stays as is (because the state is already collapsed).

After green line 148 the state is either |00> or |11>.
After green line 149 the state is either |00> or |10>.
After green line 150 the state of q[0] is either |+> or |->, which I interpret as equally probable |0> or |1> in PauliZ basis (non-collapsed, non-measured state).

The subsequent `sim->ReleaseQubit(q[0])` should fail.

Reply:
These lines are performing the adjoint of the operations above, such that the qubits are returned to the ground state, ie: |00⟩. Now that the QIR simulator wrapper checks for this condition the tests would fail with an exception; these same tests would already have failed that way if performed in the C# simulator wrapper.

Reply:
Thank you very much for explaining. 
Looks like I'm the only one who needs comments in this code ;-)